Tests/SwiftSyntaxTest/SyntaxCollectionsTests.swift

```diff
@@ -146,4 +146,32 @@ public class SyntaxCollectionsTests: XCTestCase {
     XCTAssertEqual("\(relems[1])", "1")
     XCTAssertEqual("\(relems[0])", "2")
   }
+
+  public func testFilter() {
+    let elements = ArrayElementListSyntax([
+      integerLiteralElement(0),
+      integerLiteralElement(1),
+      integerLiteralElement(2),
+    ])
+    let array = ArrayExprSyntax(elements: elements)
+
+    let filteredElements = array.elements.filter { _ in true }
+    // The filtered sequence should have the `ArrayExprSyntax` as a parent
+    XCTAssert(filteredElements.parent?.is(ArrayExprSyntax.self) ?? false)
+  }
+
+  public func testAddingSequence() {
+    let elements = ArrayElementListSyntax([
+      integerLiteralElement(0),
+      integerLiteralElement(1),
+      integerLiteralElement(2),
+    ])
+    let array = ArrayExprSyntax(elements: elements)
+
+    let addedElements: ArrayElementListSyntax = Array([integerLiteralElement(4)]) + array.elements
+    XCTAssertEqual(addedElements.count, 4)
+    // If the first collection of + doesn’t have a parent, it doesn’t really
+    // make sense to inherit the parent from the second collection.
+    XCTAssert(addedElements.parent?.is(ArrayExprSyntax.self) ?? false)
```

Comment:
I wouldn't expect this. This is basically prepending and I think I'd want to keep the parent in that case?

Reply:
`testPrependingElement` tests this case FWIW

Reply:
That's... true 🤔 The comment seems to be suggesting the opposite doesn't it? And then I read the assert as an `XCTAssertFalse`.

Reply:
Oh, the comment was outdated…